tiny_dnn/parameter.h

```diff
@@ -0,0 +1,119 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/util/util.h"
+
+namespace tiny_dnn {
+
+enum class param_type : int8_t { weight = 0x0001, bias = 0x0002 };
+
+class parameter : public std::enable_shared_from_this<parameter> {
```

Comment:
I had an idea of optimizer catching hold of all the parameters of a network while training, again similar to how pytorch performs updates ( _the `optim.step()` way_ ).